utils/check-links.pl

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/perl
+
+# Copyright 2024 Viktor Söderqvist <viktor.soderqvist@est.tech>
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause#
+
+# Checks for intenal broken links. Checks some other things too, like orphan
+# pages (nobody links to the orphan page) and the number of external links.
+
+use warnings;
+use strict;
+
+my $dry = 1; # dry run (enable for debugging)
+
+my $root = __FILE__ =~ s![^/]*$!!r . "../";
+$root =~ s%[^/\.]+/\.\./$%%;
+my @files = glob("${root}{topics,commands}/*");
+
+my %targets;
+for (@files) {
+    s/^$root//;
+    $targets{$_} = 0;
+}
+
+my %generated_targets = ("commands/" => 0,
+                         "clients/" => 0,
+                         "modules/" => 0);
+
+my ($num_abs_url, $num_abs_path, $num_broken, $num_valid, $num_generated) = (0,0,0,0,0);
+for (@files) {
+    my $filename = $root . $_;
+    next unless m!([^/]*)/([^/]*\.md)$!;
+    my ($dir, $basename) = ($1, $2);
+
+    #print "$dir / $basename\n";
+    #exit;
```

Comment:
More dead code. I don't understand this file at all though, so just going to assume it's all right.

Reply:
Hehe, Perl is not your mother toungue? ;) I wasn't going to add this in the same PR, but I must have added it. Later, I think we can add a CI job to check for broken internal links. It works and I've found and fixed some links in earlier doc PRs, without sharing the script.

I'll delete the dead code.